web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java

```diff
@@ -137,6 +137,40 @@ public String getDatastreamPid(ContentObjectRecord briefObj, AccessGroupSet prin
         return null;
     }
 
+    /**
+     * Gets first object with an original file to use as thumbnail
+     * Returns null if briefObject has an original file, as we don't need to add the object twice.
+     * @param briefObj
+     * @param principals
+     * @return
+     */
+    public ContentObjectRecord getThumbnailObject(ContentObjectRecord briefObj, AccessGroupSet principals) {
+        if (permissionsHelper.hasOriginalAccess(principals, briefObj)) {
+            return null; // Already have briefObj for thumbnail. Don't need it again
+        }
+
+        // Check if there's only one file
+        ContentObjectRecord contentObj = getChildFileObject(briefObj, principals);
+        if (contentObj != null && permissionsHelper.hasOriginalAccess(principals, contentObj)) {
+            return contentObj;
+        }
+
+        // Check other children for thumbnails
+        QueryResponse childContentObjs = getChildFileObjects(briefObj, principals, MAX_FILES);
```

Comment:
Instead of querying for all the children, I believe you can execute a solr query that limits to files which have one of the thumbnail datastreams and which the user has permission to view. For query based on the presence of a particular datastream, see:
https://github.com/UNC-Libraries/box-c/blob/main/web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java#L227

For the permission restriction, the readGroup field will actually tell you this since it reflects if the user has viewMetadata permissions, so simply doing a normal permission check (adding the principals to a search request) should be sufficient in the case of thumbnails. Might want to add a comment to the method to make that clear though.

I believe this query could be limited to 1 result, and it would also double for the case where there is only one child, so you might be able to remove that part of the method.